File: AAS.Domain/Users/Roles/UserRoleBlank.cs
#region

using AAS.Domain.AccessPolicies;
using AAS.Tools.Types.IDs;

#endregion

namespace AAS.Domain.Users.Roles;

public class UserRoleBlank
{
    public ID? Id { get; set; }
    public string? Name { get; set; }
    public AccessPolicy[] AccessPolicies { get; set; }

    public UserRoleBlank(ID? id, string? name, AccessPolicy[] accessPolicies)
    {
        Id = id;
        Name = name;
        AccessPolicies = accessPolicies;
    }
}

File: src/AAS.BackOffice/client_app/src/domain/users/roles/userRoleBlank.ts
import {AccessPolicy} from "../../accessPolicies/accessPolicy";
import {UserRole} from "./userRole";

export interface UserRoleBlank {
    id: string | null,
    name: string | null,
    accessPolicies: AccessPolicy[]
}

export namespace UserRoleBlank {
    export function getDefault(): UserRoleBlank {
        return {
            id: null,
            name: null,
            accessPolicies: []
        }
    }

    export function fromUserRole(userRole: UserRole): UserRoleBlank {
        return {
            id: userRole.id,
            name: userRole.name,
            accessPolicies: userRole.accessPolicies
        }
    }
}